open-metadata-implementation/access-services/data-engine/data-engine-server/src/test/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineConnectionAndEndpointHandlerTest.java

```diff
@@ -0,0 +1,126 @@
+import static org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIMapper.CSV_FILE_TYPE_NAME;
+import static org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIMapper.ENDPOINT_TYPE_NAME;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.WARN)
+class DataEngineConnectionAndEndpointHandlerTest {
+
+    private static final String USER = "user";
+    private static final String METHOD = "method";
+    private static final String ASSET_QUALIFIED_NAME = "qualifiedName";
+    private static final String EXTERNAL_SOURCE_GUID = "externalSourceGuid";
+    private static final String EXTERNAL_SOURCE_NAME = "externalSourceName";
+    private static final String PROTOCOL = "protocol";
+    private static final String NETWORK_ADDRESS = "networkAddress";
+    private static final String CONNECTION_QUALIFIED_NAME = CONNECTION_TYPE_NAME + "::" + PROTOCOL + "::" + NETWORK_ADDRESS ;
+    private static final String ENDPOINT_QUALIFIED_NAME = ENDPOINT_TYPE_NAME + "::" + PROTOCOL + "::" + NETWORK_ADDRESS ;
+
+    @Mock
+    private OMRSRepositoryHelper repositoryHelper;
+    @Mock
+    private InvalidParameterHandler invalidParameterHandler;
+    @Mock
+    private DataEngineCommonHandler dataEngineCommonHandler;
+    @Mock
+    private ReferenceableHandler<Connection> connectionHandler;
+    @Mock
+    private ReferenceableHandler<Endpoint> endpointHandler;
+
+    @InjectMocks
+    private DataEngineConnectionAndEndpointHandler dataEngineConnectionAndEndpointHandler;
+
+    @Test
+    void insertConnectionAndEndpoint() throws InvalidParameterException, PropertyServerException, UserNotAuthorizedException {
+        mockDataEngineCommonHandler(true);
+
+        dataEngineConnectionAndEndpointHandler.upsertConnectionAndEndpoint(ASSET_QUALIFIED_NAME, CSV_FILE_TYPE_NAME, PROTOCOL,
+                NETWORK_ADDRESS, EXTERNAL_SOURCE_GUID, EXTERNAL_SOURCE_NAME, USER, METHOD);
+
+        verify(dataEngineCommonHandler, times(1)).findEntity(USER, ASSET_QUALIFIED_NAME, CSV_FILE_TYPE_NAME);
+
+        verify(dataEngineCommonHandler, times(1)).findEntity(USER, CONNECTION_QUALIFIED_NAME, CONNECTION_TYPE_NAME);
+//        verify(connectionHandler, times(1)).createBeanInRepository(eq(USER), eq(EXTERNAL_SOURCE_GUID),
+//                eq(EXTERNAL_SOURCE_NAME), eq(CONNECTION_TYPE_GUID), eq(CONNECTION_TYPE_NAME), eq(CONNECTION_QUALIFIED_NAME),
+//                eq(QUALIFIED_NAME_PROPERTY_NAME), any(), eq(METHOD));
+        verify(dataEngineCommonHandler, times(1)).upsertExternalRelationship(eq(USER), any(), any(),
+                eq(CONNECTION_TO_ASSET_TYPE_NAME), eq(CONNECTION_TYPE_NAME), eq(EXTERNAL_SOURCE_NAME), any());
+
+        verify(dataEngineCommonHandler, times(1)).findEntity(USER, ENDPOINT_QUALIFIED_NAME, ENDPOINT_TYPE_NAME);
+//        Commented verification cannot be performed because of mockito bug: unable to properly inject generics
```

Comment:
You should be able to verify this method if you use a mocked builder instead of any() - you can check DataEngineCollectionHandlerTest.createCollectionTest() for a similar example